### DISCUSSION TITLE
nginx-proxy received event die for container constantly

### DISCUSSION THREAD

**Discussion by lastb0isct** [nginx-proxy/nginx-proxy#2589] [Q&A]
My setup was working before but it looks like something must be broken now.  When starting up my container I get this at the beginning of start-up:

``` 
WARNING: /etc/nginx/dhparam/dhparam.pem was not found. A pre-generated dhparam.pem will be used for now while a new one


is being generated in the background.  Once the new dhparam.pem is in place, nginx will be reloaded.


forego     | starting dockergen.1 on port 5000


forego     | starting nginx.1 on port 5100


dockergen.1 | 2019/04/13 18:11:12 Generated '/etc/nginx/conf.d/default.conf' from 22 containers


dockergen.1 | 2019/04/13 18:11:12 Running 'nginx -s reload'


dockergen.1 | 2019/04/13 18:11:12 Watching docker events


dockergen.1 | 2019/04/13 18:11:12 Contents of /etc/nginx/conf.d/default.conf did not change. Skipping notification 'nginx -s reload'


dockergen.1 | 2019/04/13 18:11:22 Received event stop for container 6215da439c5c


dockergen.1 | 2019/04/13 18:11:22 Contents of /etc/nginx/conf.d/default.conf did not change. Skipping notification 'nginx -s reload'


dockergen.1 | 2019/04/13 18:12:06 Received event start for container 20e32ec4ef54 
```

Then a couple minutes later I get this:

```
2019/04/13 18:13:20 [notice] 82#82: signal process started


Generating DH parameters, 2048 bit long safe prime, generator 2


This is going to take a long time


dhparam generation complete, reloading nginx


dockergen.1 | 2019/04/13 18:14:01 Received event start for container 20e32ec4ef54


dockergen.1 | 2019/04/13 18:14:01 Contents of /etc/nginx/conf.d/default.conf did not change. Skipping notification 'nginx -s reload'


dockergen.1 | 2019/04/13 18:14:02 Received event die for container 20e32ec4ef54


dockergen.1 | 2019/04/13 18:14:02 Contents of /etc/nginx/conf.d/default.conf did not change. Skipping notification 'nginx -s reload'


dockergen.1 | 2019/04/13 18:15:02 Received event start for container 20e32ec4ef54
```

This continues forever.  It is not reachable from the ports that are presented to it which shows nginx isn't even starting.  Any help would be appreciated.  This is my config:

```
docker run -d -p 80:80 -p 443:443 \
    --name nginx-proxy \
    --net reverse-proxy \
    -v /etc/nginx/certs \
    -v vhost:/etc/nginx/vhost.d \
    -v html:/usr/share/nginx/html \
    -v /var/run/docker.sock:/tmp/docker.sock:ro \
    --restart=unless-stopped \
    --label com.github.jrcs.letsencrypt_nginx_proxy_companion.nginx_proxy=true \
    jwilder/nginx-proxy
```

**Comment by lastb0isct** [comment]
Any ideas/help?

**Comment by Greek64** [comment]
You get the `Received event die` and `Received event start` log messages whenever ANY docker container on the system is started or stopped.
That means that somewhere a container is failing to start, crashes and restarts (probably due to a --restart flag).  
This should not have to do anything with `nginx-proxy` in particular. I would use `docker ps -a` to see all running/stopped containers and try to find the culprit.

This however (the starting/stopping of other containers) should not affect the `nginx-proxy` in any way beyond reloading it's configuration.  
So since you state that it's not accessible I guess the container that fails to start and restarts is the actual container you try to reverse proxy and access.

**Comment by PetterTables** [comment]
docker inspect --format='{{.LogPath}}'  ID_CONTAINER

and check log

**Comment by carloscoronad0** [comment]
Don't know if it's too late, but better late than never. Maybe you are binding the docker socket `/var/run/docker.sock` to the nginx container. I think all containers share the socket, so that may be a hint on why other containers are being affected.

**Comment by benjaoming** [comment]
I just discovered this very issue.

Turns out I had a restart loop, because two containers were competing for the same local port with `--restart=always`.

The restart loop looks like this:

```
❯ docker ps 
CONTAINER ID   IMAGE           COMMAND                  CREATED        STATUS                                    PORTS                                     NAMES
c3649073c51c   acme:latest   "/docker-entrypoint.…"   3 weeks ago    Up 2 minutes                              0.0.0.0:1234->80/tcp, [::]:1234->80/tcp   heuristic_moore
6e729d24e155   9fcfd17716ef    "/docker-entrypoint.…"   5 months ago   Restarting (137) Less than a second ago                                             thirsty_black
```

Killing the container was also very difficult, but using `docker rm -f` seemed to finally get it to stop restarting.

**Comment by benjaoming** [comment]
Some suggestions:

1) Detect loops in "Received event start for container <id>" by counting the number of times that a container is causing reloads.

2) Add suggestions to the message "Received event start for container <id> - monitor your containers with for instance `watch docker ps` locally to see details."

3) Move this issue to Discussions since it's not strictly speaking related to a bug. Otherwise, mark it as a feature proposal.
